File: src/main/java/org/kite9/diagram/model/position/RectangleRenderingInformationImpl.java
package org.kite9.diagram.model.position;

import org.apache.commons.math.fraction.BigFraction;
import org.kite9.diagram.common.objects.OPair;

public class RectangleRenderingInformationImpl extends AbstractRenderingInformationImpl implements RectangleRenderingInformation {
	
	private OPair<BigFraction> gx, gy;
	private int sx, sy;
	private double[] xc, yc;
	
	public RectangleRenderingInformationImpl() {
		super();
	}
	
	public RectangleRenderingInformationImpl(Dimension2D pos, Dimension2D size, Direction orientation, boolean rendered) {
		setPosition(pos);
		setSize(size);
		setRendered(rendered);
	}

	@Override
	public OPair<BigFraction> gridXPosition() {
		return gx;
	}

	@Override
	public OPair<BigFraction> gridYPosition() {
		return gy;
	}

	public void setGridXPosition(OPair<BigFraction> gx) {
		this.gx = gx;
	}
	
	public void setGridYPosition(OPair<BigFraction> gy) {
		this.gy = gy;
	}

	@Override
	public int gridXSize() {
		return sx;
	}

	@Override
	public int gridYSize() {
		return sy;
	}

	@Override
	public void setGridXSize(int x) {
		this.sx = x;
	}

	@Override
	public void setGridYSize(int y) {
		this.sy = y;
	}

	@Override
	public double[] getCellXPositions() {
		return xc;
	}

	@Override
	public double[] getCellYPositions() {
		return yc;
	}

	@Override
	public void setCellXPositions(double[] xs) {
		this.xc = xs;
	}

	@Override
	public void setCellYPositions(double[] ys) {
		this.yc = ys;
	}
}

File: src/main/kotlin/org/kite9/diagram/model/position/RectangleRenderingInformationImpl.kt
package org.kite9.diagram.model.position

import org.kite9.diagram.model.position.AbstractRenderingInformationImpl
import org.kite9.diagram.model.position.RectangleRenderingInformation
import org.kite9.diagram.common.objects.OPair
import org.kite9.diagram.common.fraction.BigFraction

class RectangleRenderingInformationImpl : AbstractRenderingInformationImpl, RectangleRenderingInformation {

    private var gx: OPair<BigFraction>? = null
    private var gy: OPair<BigFraction>? = null
    private var sx = 0
    private var sy = 0
    override var cellXPositions: DoubleArray? = doubleArrayOf()
    override var cellYPositions: DoubleArray? = doubleArrayOf()

    constructor() : super() {}
    constructor(pos: Dimension2D?, size: Dimension2D?, rendered: Boolean) {
        this.position = pos
        this.size = size
        this.rendered = rendered
    }

    override fun gridXPosition(): OPair<BigFraction>? {
        return gx
    }

    override fun gridYPosition(): OPair<BigFraction>? {
        return gy
    }

    override fun setGridXPosition(gx: OPair<BigFraction>?) {
        this.gx = gx
    }

    override fun setGridYPosition(gy: OPair<BigFraction>?) {
        this.gy = gy
    }

    override fun gridXSize(): Int {
        return sx
    }

    override fun gridYSize(): Int {
        return sy
    }

    override fun setGridXSize(x: Int) {
        sx = x
    }

    override fun setGridYSize(y: Int) {
        sy = y
    }
}